ens/main.py

```diff
@@ -0,0 +1,317 @@
+
+import idna
+
+from ens import abis
+from ens.constants import EMPTY_SHA3_BYTES
+from ens.exceptions import (
+    AddressMismatch,
+    InvalidName,
+    UnauthorizedError,
+    UnownedName,
+)
+from ens.registrar import Registrar
+from ens.utils import dict_copy, ensure_hex, init_web3
+
+DEFAULT_TLD = 'eth'
+RECOGNIZED_TLDS = [DEFAULT_TLD, 'reverse', 'test']
+
+ENS_MAINNET_ADDR = '0x314159265dd8dbb310642f98f50c066173c1259b'
+
+REVERSE_REGISTRAR_DOMAIN = 'addr.reverse'
+
+GAS_DEFAULT = {
+    'setAddr': 60001,
+    'setResolver': 60002,
+    'setSubnodeOwner': 60003,
```

Comment:
Actually, I can remove all the gas limits (I wrote this before I knew that web3 auto-estimates).  I'll leave in the start-auction limits, because it's helpful to warn people when the block gets too big.

At some point, it seems like that could/should be replaced by a generalized solution when estimating the gas. I'll think about that and make a note to open a separate PR for it.